src/Umbraco.Core/Persistence/Factories/PropertyFactory.cs

```diff
@@ -131,6 +133,15 @@ public static IEnumerable<PropertyDataDto> BuildDtos(int currentVersionId, int p
                         {
                             editedCultures.Add(propertyValue.Culture); // report culture as edited
                         }
+
+                        // flag culture as edited if it contains an edited invariant property 
+                        if (propertyValue.Culture == null &&
```

Comment:
each of these lines should have comments as to what they do just like the code block above since all this code is quite hard to follow